src/scipp/plt/model.py

```diff
@@ -0,0 +1,142 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .widgets.widget import WidgetView
+
+from typing import Tuple, Iterable
+from functools import partial
+
+
+class Node:
+
+    def __init__(self, func, name=None, views=None):
+        self.name = name
+        self.parent_name = None
+        self.dependency = None
+        self.func = func
+        self.views = []
+        if views is not None:
+            for view in views:
+                self.views.append(view)
+
+    def send_notification(self, message):
+        # print(f'hey there! from {self.name}: ', message)
+        for view in self.views:
+            view.notify({
+                "node_name": self.name,
+                "parent_name": self.parent_name,
+                "message": message
+            })
+
+    def request_data(self):
+        pipeline = []
+        node = self
+        while node.dependency is not None:
+            pipeline.append(node.func)
+            node = node.dependency
+        else:
+            pipeline.append(node.func)
+        pipeline = list(pipeline[::-1])
```

Comment:
Why do you call `list` here?

Reply:
I'm not sure, I remember running into issues with in-place access and that I needed to make a copy, but there was probably something else involved.
As far as I can tell, as it stands now, it should be fine with just `pipeline = pipeline[::-1]`